sn_networking/src/event.rs

```diff
@@ -834,7 +834,9 @@ impl SwarmDriver {
             } => {
                 event_string = "kad_event::RoutingUpdated";
                 if is_new_peer {
-                    info!("New peer added to routing table: {peer:?}");
+                    // todo: cache the count as it is frequently used during replication as well.
+                    let connected_peers = self.swarm.connected_peers().count();
+                    info!("New peer added to routing table: {peer:?}, now we have #{connected_peers} connected peers");
```

Comment:
If we want to avoid this being removed again, let's make this a log marker (or whatever the equivalent is now)

Reply:
I've updated the PR to now cache the `connected_peers` instead.